### PR TITLE
chore: enable insider preview by default

### PR DESCRIPTION
**Pull request by a1exwang**
- remove insider preview from vscode settings
- change `TEAMSFX_INSIDER_PREVIEW` env variable to `__TEAMSFX_INSIDER_PREVIEW` (this env var is only for unit test and other testings. It is not exposed to end users). It defaults to true if not set. This is to prevent users that have `TEAMSFX_INSIDER_PREVIEW` set from being affected.
- Change unit-test CI accordingly
- Update docs accordingly

### CODE REVIEW

**Comment by a1exwang** [comment]
E2E TEST: https://github.com/OfficeDev/TeamsFx/blob/dev/packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
           # make sure the coverage data is cleaned
           rm -rf .nyc_output
           xvfb-run -a npx lerna run test:unit --stream
-          TEAMSFX_INSIDER_PREVIEW=true xvfb-run -a npx lerna run test:unit --stream
+          __TEAMSFX_INSIDER_PREVIEW=false xvfb-run -a npx lerna run test:unit --stream
```
Comment (IvanJobs) [inline .github/workflows/unit-test.yml:47]: It seems some parts are missing for cicd feature.
Can you help remove the non-insider version of cicd support in this PR too? Thanks.
1. remove folder [/docs/cicd](https://github.com/OfficeDev/TeamsFx/tree/dev/docs/cicd)
2. remove non-insider `cicd guide` button, refer to code: [L505](https://github.com/OfficeDev/TeamsFx/blob/15f361caa3c88439dc4c9c324090baf01f27a189/packages/vscode-extension/src/handlers.ts#L505)
3. update readme.md refer to [readme.md](https://github.com/OfficeDev/TeamsFx/blob/dev/docs/cicd_insider/README.md)
image

**Review by jayzhang — APPROVED** [review]

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (a1exwang) [reply .github/workflows/unit-test.yml:47]: Thanks, I will update it.

**Review by nliu-ms — APPROVED** [review]

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (a1exwang) [reply .github/workflows/unit-test.yml:47]: Offline discussed. 1 and 2 are code cleanup work, and it will be in the next sprint. 3 is updated.

**Review by IvanJobs — APPROVED** [review]
lgtm for the CICD part.

**Review by qinezh — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by 1openwindow — APPROVED** [review]

**Review by Alive-Fish — APPROVED** [review]
This PR may block some e2e test cases and talked with @a1exwang, he will handle the potential failures.

**Pull request merged**